6-infrastructure-configuration-management/6.1.3-terraform-modules.md

```diff
@@ -4,6 +4,144 @@ The previous sections should have illustrated how flexible and powerful Terrafor
 
 Read Hashicorp's [Terraform module overview](https://developer.hashicorp.com/terraform/tutorials/modules/module) to learn more about Terraform modules features and uses.
 
+ As mentioned, there are registry and remote modules. Both types of modules are below. By this point throughout the Terraform chapters, you may or may not have a VPC in the AWS account you are deploying to. The first exercise creates an ec2 instance and VPC in AWS. Alternatively you may want to add an EC2 instance to an existing VPC, that is also covered below. 
+
+
 ## Exercise
 
-TODO
+### **Registry Modules** 
+
+There are a multitude of registry modules available. For this exercise the modules used are managed my hashicorp. 
+
+?> Whenever you add a new module to a configuration, Terraform must install the module before it can be used. Both the `terraform get` and `terraform init` commands will install and update modules.
+
+1. Review the configuration first. The `  source  = "terraform-aws-modules/vpc/aws"
+` and `  source  = "terraform-aws-modules/ec2-instance/aws"
+` declare the registry and module thats being used for the modules in this configuration.
+
+2. Create a new `.tf` file or add the following configuration to your `main.tf` file from previous sections of the boot camp.  
+
+3. For this exercise the `ami` id is hardcoded. A new ID can be found in the AWS console or the AWS CLI. This AWS user guide walks through that process - [Find a Linux AMI](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/finding-an-ami.html#finding-an-ami-console)
+
+?> As mentioned previously, the configuration below creates a new VPC for the EC2 instance to reside in. If you don't want to create a new VPC, you can omit the VPC module from your configuration. For this exercise, the ec2 instance relies on outputs from the VPC module. `vpc_security_group_ids` and `subnet_id` will have to be hardcoded based on resources available in the AWS account you're deploying into. 
+
+
```

Comment:
In the intro it talks about how we are going to be either creating a new vpc or attaching it to an existing one but here it seems we are only attaching to an existing one. We are also grabbing variables from a file we havn't explicitly created which seems to indicate we are just missing something rather than having those variables be something we have to input ourselves.

Reply:
I did skip the Var file for the registry modules, let me add those to the exercise. The two modules that I have in the exercise would create a VPC and a EC2. I'm using this note to describe a scenario in which someone could deploy just an EC2, using the EC2 module and omitting the  VPC module.